### ISSUE TITLE
pkg_tar: remap_paths should favor the longest match

### ISSUE THREAD

**Issue by guoshimin** [bazelbuild/rules_pkg#258]
Currently, remap_paths will remap each file according to the first match in the map iteration order:
https://github.com/bazelbuild/rules_pkg/blob/e8bff981f600a4faa01d5c8f5dd57d05891c8e2d/pkg/pkg.bzl#L25-L30.

This makes it impossible to have predictable results when you have two keys in remap_paths where one is a prefix of the other. I think we should favor the longest (most specific) match in that case.

**Comment by aiuto** [comment]
remap_paths is a mis-feature in the first place. Any effort should be put into generic path remapping via pkg_filegroup so that we can use this in any packaging.

**Comment by guoshimin** [comment]
Can I use pkg_filegroup as input to pkg_tar yet?

**Comment by aiuto** [comment]
No. I guess I left that unsaid. With limited resources, I'm not going to spend any of my time working on things that are obsolete.
